### PR TITLE
 Feature: auction extension period

### PR DESCRIPTION
**Pull request by iorveth**
Closes https://github.com/Joystream/joystream/issues/2656,
https://github.com/Joystream/joystream/issues/2771

Depends on https://github.com/Joystream/joystream/pull/2512

### CODE REVIEW

File: runtime-modules/content/src/nft/types.rs [hunk 1]
```diff
@@ -436,14 +450,32 @@ pub struct AuctionParams<VideoId, BlockNumber, Balance, MemberId: Ord> {
...
     }
 }
+
+/// English auction details
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct EnglishAuctionDetails<BlockNumber> {
+    // the remaining time on a lot will automatically reset to to the preset extension time
+    // if a new bid is placed within that period
+    pub extension_period: BlockNumber,
```
Comment (bedeho) [inline runtime-modules/content/src/nft/types.rs:472]: Is providing a value of `0` the way you avoid having any extension at all?

**Review by bedeho — CHANGES_REQUESTED** [review]
- Am I correct that we are still automatically closing auctions if a buy now bid is made, even if its within the extension window? I think that is the most appropriate.
_ I think it would be useful to clearly signal in the event `AuctionBidMade`  when the auction is extended.

File: runtime-modules/content/src/nft/types.rs (hunk 1 above)
Comment (iorveth) [reply runtime-modules/content/src/nft/types.rs:472]: Yes, it's correct and possible in theory.
Though we also have safety bounds for both `English` `auction_duration` and `extension_period` parameters.

Comment (iorveth) [reply runtime-modules/content/src/nft/types.rs:472]: I added additional safety check here as well https://github.com/Joystream/joystream/pull/2667/commits/5e271e3a29aac70b7be46f23eb83570c8286e4e3

**Comment by iorveth** [comment]
> * Am I correct that we are still automatically closing auctions if a buy now bid is made, even if its within the extension window? I think that is the most appropriate.
>
You are correct, this features don't cause any conflict with each other 
>   _ I think it would be useful to clearly signal in the event `AuctionBidMade`  when the auction is extended.
>
Done here https://github.com/Joystream/joystream/pull/2667/commits/d6f95b93b126a3cfbcfa7b34d11859360e7e0779

**Review by bedeho — APPROVED** [review]
Looks good!

**Pull request merged**